clasi_user/interface.py

```diff
@@ -0,0 +1,173 @@
+            else:
+                if not os.path.exists(one_file_path.parent/'csv'):
+                    os.makedirs(one_file_path.parent/'csv')
+
+            if test: dest = one_file_path.parent/'delete_test'
+            else: dest = one_file_path.parent/'csv'
+            filename = Path(one_file_path.name).stem + '.csv'
+            df.to_csv(dest/filename,index=False)
+            print(f'convert {filename} to csv')
+        self.csv_path = dest
+
+    def _prepare_df(self):
+        dfs = []
+        for file in self.csv_path.ls():
+            df = pd.read_csv(file)
+            dfs.append(df)
+        return pd.concat(dfs)
+
+    def _train_valid_split(self,total_filenames,train_pct=0.7):
+        '''split the train and test base on total datasets
+            return train_list and valid_list with their associated path'''
+        total_filenames = total_filenames.ls()
+        random.seed(42)
+        random.shuffle(total_filenames)
+
+        split_point = round(len(total_filenames)*train_pct)
+        assert (type(split_point) == int)
+        self.df_train, self.df_valid = total_filenames[:split_point],total_filenames[split_point:]
+
+    def _get_lm(self,url):
+        '''This code needs to be updated once I figure out how to host pretrained weights
+            for now, it will use local path
+        '''
+        if url:
+            data_lm = load_data(url,'data_lm_large.pkl',bs=self.bs)
+        return data_lm
+
+    def _get_individual_data(self,filepath,data_lm):
+        df = pd.read_csv(filepath)
+        if df.shape[0]*0.7 < self.bs:
+            bs = int(df.shape[0]*0.7)
+        else: bs=self.bs
+        data = (TextList
+                .from_df(df=df,path=self.csv_path,cols='Body',vocab=data_lm.vocab)
+                .split_by_rand_pct(0.3,seed=42)
+                .label_from_df(cols='Label')
+                .databunch(bs=bs,num_workers=os.cpu_count()*4)
+               )
+        return data
+
+
+    def pre_processing(self,skip_convert_json=False,test=False,**kwargs):
+        if not self._check_data_format(): exit(0) #we can't handle non Basilica data
+        if not skip_convert_json: self._convert_json_to_csv(test)
+        if self.eval_mode:
+            if 'lm_path' in kwargs.keys():
+                lm_path = kwargs['lm_path']
+            else:
+                print('Evaluation mode: lm_path not set...')
+                exit(0)
+            data_lm = self._get_lm(lm_path)
+            for file in self.csv_path.ls():
+                file_extension = Path(file).name.split('.')
+                if len(file_extension) < 2 or file_extension[1] != 'csv': continue
+                self.data_list.append(self._get_individual_data(file,data_lm))
+                self.dataset_name.append(file_extension[0])
+        else:
+            '''
+            Getting language model data bunch
+            Split general classifier train and valid
+            '''
```

Comment:
Docstring does not belong here. Either change this to multiline comment or move it to the first statement of the function. See here for examples: https://www.techbeamers.com/understand-python-comment-docstring/